### PR TITLE
Support view type for creating new untitled file.

### PR DESCRIPTION
**Pull request by rebornix**
The PR makes opening an untitled custom/notebook editor possible. It includes two sets of changes

* Move `GlobalNewUntitledFileAction` to command `NEW_UNTITLED_FILE_COMMAND_ID `
  * `registerWorkbenchAction` is replaced by `MenuRegistry.addCommand`,  `MenuRegistry.appendMenuItem ` and `appendToCommandPalette`. @bpasero can you confirm if it's all I need to do?
* Add a second parameter `viewType` for the command. This one is straightforward and doesn't affect existing `Ctrl/Cmd+N` experience.

### CODE REVIEW

File: src/vs/workbench/contrib/files/browser/fileActions.contribution.ts [hunk 1]
```diff
@@ -42,11 +42,16 @@ registry.registerWorkbenchAction(SyncActionDescriptor.from(FocusFilesExplorer),
 registry.registerWorkbenchAction(SyncActionDescriptor.from(ShowActiveFileInExplorer), 'File: Reveal Active File in Side Bar', category.value);
 registry.registerWorkbenchAction(SyncActionDescriptor.from(CollapseExplorerView), 'File: Collapse Folders in Explorer', category.value);
 registry.registerWorkbenchAction(SyncActionDescriptor.from(RefreshExplorerView), 'File: Refresh Explorer', category.value);
-registry.registerWorkbenchAction(SyncActionDescriptor.from(GlobalNewUntitledFileAction, { primary: KeyMod.CtrlCmd | KeyCode.KEY_N }), 'File: New Untitled File', category.value);
 registry.registerWorkbenchAction(SyncActionDescriptor.from(CompareWithClipboardAction, { primary: KeyChord(KeyMod.CtrlCmd | KeyCode.KEY_K, KeyCode.KEY_C) }), 'File: Compare Active File with Clipboard', category.value);
 registry.registerWorkbenchAction(SyncActionDescriptor.from(ToggleAutoSaveAction), 'File: Toggle Auto Save', category.value);
 registry.registerWorkbenchAction(SyncActionDescriptor.from(ShowOpenedFileInNewWindow, { primary: KeyChord(KeyMod.CtrlCmd | KeyCode.KEY_K, KeyCode.KEY_O) }), 'File: Open Active File in New Window', category.value);
 
+MenuRegistry.addCommand({
```
Comment (bpasero) [inline src/vs/workbench/contrib/files/browser/fileActions.contribution.ts:49]: This is a bit confusing to me, shouldn't this move to the `// Commands` section and be `CommandsRegistry.registerCommand`?

Comment (bpasero) [reply src/vs/workbench/contrib/files/browser/fileActions.contribution.ts:49]: On the other hand, isn't this command already registered via `KeybindingsRegistry.registerCommandAndKeybindingRule`

**Review by bpasero — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/files/browser/fileActions.contribution.ts [hunk 2]
```diff
@@ -61,6 +60,11 @@ if (isMacintosh) {
 // Commands
 CommandsRegistry.registerCommand('_files.windowOpen', openWindowCommand);
 CommandsRegistry.registerCommand('_files.newWindow', newWindowCommand);
+MenuRegistry.addCommand({
```
Comment (rebornix) [inline src/vs/workbench/contrib/files/browser/fileActions.contribution.ts:63]: @bpasero @sbatten 

I registered this command to MenuRegistry as it's part of `registerWorkbenchAction`, which consists of 

* `CommandsRegistry.registerCommand`
* `KeybindingsRegistry.registerKeybindingRule`
* `MenuRegistry.addCommand`
* `MenuRegistry.appendMenuItem`

I'm not 100% sure of the intention of holding a cache of commands in the MenuRegistry and `MenuRegistry.getCommand` is used in quite a few places.

**Review by bpasero — CHANGES_REQUESTED** [review]
I would remove the call to `MenuRegistry.addCommand` as I do not understand it. I also think that Jo has a way to contribute actions all in one go. 

https://github.com/microsoft/vscode/blob/1cc95bad5440a0ce78db0bf0d2913f3f48325166/src/vs/platform/actions/common/actions.ts#L461

**Comment by rebornix** [comment]
@bpasero I removed the menu command registration for now after discussing with @sbatten. Moving to `registerAction2` can be a debt item later on since it involves more changes in the `workbench/files`.

**Review by bpasero — APPROVED** [review]

**Pull request merged**